compiler/rustc_lint/src/nonstandard_style.rs

```diff
@@ -57,7 +57,7 @@ declare_lint! {
 declare_lint_pass!(NonCamelCaseTypes => [NON_CAMEL_CASE_TYPES]);
 
 fn char_has_case(c: char) -> bool {
-    c.is_lowercase() || c.is_uppercase()
+    c.to_lowercase().to_string() != c.to_uppercase().to_string()
```

Comment:
This could be written with smaller generated code with https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=db8dc3325d0f3b2cf75bb28eb62f33fd

https://rust.godbolt.org/z/sdqYGv